### ISSUE TITLE
Mac M1 - Pi Imager 1.7.1 and 1.6.2 not working

### ISSUE THREAD

**Issue by SISLANGER** [raspberrypi/rpi-imager#398]
Hi, I cant get the Pi Imager to burn an SD correctly on Mac Mini M1 (Monetery).

I goes through the process but at the end it says "Error reading from storage. SD card may be broken"

I then have togo into disk utility to reformat the flash drive to get it working again. The drive is fine and so is the source file, they burn correctly in balenaEtcher. But I much prefer Pi Imager.

1.6.2 worked fine until I updated to 1.7.1 now neither work.

I have even given the Pi Imager all disk access and still no luck. Any suggestions

**Comment by maxnet** [comment]
>The drive is fine and so is the source file

Still suggest you try the standard RPI OS image file, and a different drive, just to see if that does work.

Source file concerns Kali?

**Comment by SISLANGER** [comment]
@maxnet if I select an image either Raspberry Pi OS or any other via the menu, it gives the completed message but then the SD cant be read by the computer and will not boot in a Pi.

**Comment by maxnet** [comment]
>@maxnet if I select an image either Raspberry Pi OS or any other via the menu, it gives the completed message but then the 
>SD cant be read by the computer and will not boot in a Pi.

Tried a different SD card and USB SD card reader?


It does is an external reader, right?
As someone else did report the internal SD card reader -that is only available on higher-end Macbook Pro models- being temperamental.

**Comment by maxnet** [comment]
>Are there caches, libraries, config files in folders that could be causing the issue ?

Anything that intercepts reads and writes such as anti-virus software can cause problems if it has a bug, although that usually has different symptoms.

===

> and it does the same on a USB flash drive.

If with the same you do not mean the "Error reading from storage. SD card may be broken" error, but that the Pi will not boot from the USB drive, that may have different causes.
May not boot from USB by default if it has old EEPROM firmware.
The bootloader screen does show that it attempts to boot from USB?

>A point of note, when using a USB drive, it always fails on the first admin password authentification. 

Cannot imagine anything that would cause that.

**Comment by SISLANGER** [comment]
@maxnet I did have anti-virus installed but removed because of issues with background net uses on some apps.

Pi boots fine from USB (SD card slot broken) and SD card tested in Pi Zero 2 W

Bootloader not sure, as not connected to monitors, but they hang doing nothing. 
Whereas the same file flashed in Etcher boots straight through and connects to local network with no issue.

**Comment by maxnet** [comment]
>not connected to monitors

So while you mentioned "Pi will not boot" earlier, you have no way to be sure of how far it came in the boot process, and are only drawing that conclusion because you did not see it showing up in network?

Did you connect the Pi with a network cable, or through wifi?
If the last, then it could be that the write of the main image itself did succeed, and the Pi does boot, but customizing the image with your wifi connection details goes wrong somehow.

**Comment by SISLANGER** [comment]
@maxnet I had tried previously with monitor attached, but couldnt remember if I see the rainbow startup or not.

I have checked this again attached to a screen, no rainbow boot, nothing. So even when it finally says finished somethings gone wrong.

Its strange how it can format no problem, but something is wrong with its writing process (possibly) as every now instead of failing at the end. It fails to start completly and the SD / USB is unusable until I use MacOS disk utility to reformat it. Its almost like it trying todo somehting too soon, before another command has finished.

I suspect this is one of those things that I wont solve until, I next reinstall OSX and test again. Shame though.

**Comment by glyph** [comment]
I have the same issue on an M1 Max 14" notchbook, with a CalDigit dock.

**Comment by glyph** [comment]
(I'm using the 1.7.2 imager.)

**Comment by glyph** [comment]
I worked around this by installing the imager into a Fedora 34 ARM64 VM in Parallels and passing the USB device through to the VM.

macOS version:

```
ProductName:	macOS
ProductVersion:	12.5
BuildVersion:	21G72
```

**Comment by maxnet** [comment]
>I have the same issue on an M1 Max 14" notchbook, with a CalDigit dock.

Tried a different SD card reader (preferably one that is just a SD card reader, and not a power supply/usb hub/combo device)?

**Comment by smbm** [comment]
@glyph @SISLANGER Have you tried the universal build mentioned in this issue: https://github.com/raspberrypi/rpi-imager/issues/235#issuecomment-999791765

If not it is worth a try, I and a few people I know with M1 Macs found the universal build fixed the writing issues they were having with the Rosetta build.

**Comment by glyph** [comment]
@smbm Unfortunately I've only got the one Pi, and it doesn't seem like I can easily virtualize it so it's hard to test this.  Whenever I next need to image one though, I'll be sure to use that universal build.  (Also the M1 build constantly wants to update itself to the more recent, but probably not-working Intel build :))

**Comment by bobsdavenport** [comment]
latest pi manager 1.7.3 er well?
creates password hash for wifi  even though i save actual password.

then reopening it shows the hash.
both of which dont work when sd card it booted in pi4.
meh!

**Comment by iCSpotRun** [comment]
Regardless of memory card, I still find this app unreliable on Apple Silicon. I hope the devs realize that one day Apple will sunset Rosetta 2 like they did Rosetta 1. I use it when I can, but when it breaks, I resort to alternative means of complying imagines to disks.

**Comment by tdewey-rpi** [comment]
Closing as stale.

I routinely use Raspberry Pi Imager on a Apple Silicon macBook Pro, and do not encounter reliability issues with 1.8.5.

**Comment by bjanders** [comment]
I have this problem with Raspberry Pi Imager v1.8.5, on Mac Mini M2 Pro, running MacOS 15.0.1.

I'm trying to write to a SD 32GB card using my USB 3 SD card reader.

I tried writing the Raspberry Pi OS (64-bit) image and the Retro Pi image.  As soon as I have entered root password, I get the error dialog "Error writing file to disk".

I tried ejecting and reinserting the SD card, but get the same error. I also tried restarting the app.
